[18.0][MIG] stock_warehouse_security: Migration to 18.0

christian-ramos-tecnativa opened the pull request:
Migrated from https://github.com/OCA/stock-logistics-warehouse/pull/1925.

- Module renamed from `stock_multi_warehouse_security` to `stock_warehouse_security` as the tag `multi` wasn't adding extra information about the module.
- Removed dependency to stock_warehouse_relationship

@Tecnativa TT58828

florian-dacosta committed:
[ADD] stock_warehouse_security

petrus-v committed:
[REF] stock_warehouse_security:

This is not a simple migration bettween 12.0 and 14.0
behaviour changes that users are able to only see warehouses set
on the user form or all if not set.

This module won't add more groups but based on ir.rules restrict
allowed data to see

Also we split ir.rules and the fact to add warehouse_id on models
in different modules.

florian-dacosta committed:
[IMP] stock_warehouse_security: black, isort, prettier

florian-dacosta committed:
[16][MIG] stock_warehouse_security

christian-ramos-tecnativa committed:
[IMP] stock_warehouse_security: pre-commit auto fixes

christian-ramos-tecnativa committed:
[MIG] stock_warehouse_security: Migration to 18.0

christian-ramos-tecnativa commented:
After some research, I will re-propose this module without the stock_warehouse_relationship dependency as I think it may not be needed anymore

carlos-lopez-tecnativa reviewed (changes requested):
I think it might be necessary to rewrite the entire Git history, or is it not needed, @pedrobaeza? Something like:
```
git filter-branch --tree-filter 'if [ -d old_module_name ]; then mv old_module_name new_module_name; fi' HEAD
git rebase origin/branch
```

Also, I think this module might need to invalidate the cache when the warehouse is assigned to the user. I tested it with the demo user, but on another page I can still see all records (pickings, operation types, etc.).

<img width="871" height="791" alt="image" src="https://github.com/user-attachments/assets/2e23611a-607a-4ead-9269-9041600700a4" />

<img width="1009" height="368" alt="image" src="https://github.com/user-attachments/assets/57dbd2d7-f479-4397-a568-687d5b7e3ddf" />

<img width="1744" height="745" alt="image" src="https://github.com/user-attachments/assets/cd7b09b7-ea8a-40b9-9aa2-f391f99d8f77" />

I restarted the Runboat instance to see the changes:
<img width="1744" height="745" alt="image" src="https://github.com/user-attachments/assets/e7417530-553f-41bf-bc22-5ef90e6a2eb6" />

pedrobaeza commented:
Yes, it would be good to rewrite whole history.

christian-ramos-tecnativa commented:
I found that when a warehouse is selected, if any product has stock in a location belonging to a different warehouse, you can no longer access the product view.
This issue seems to be related to how `qty_available` is computed and will most likely require some adjustments.
I’ll provide an update as soon as I check it.

christian-ramos-tecnativa commented:
Thanks for the review @carlos-lopez-tecnativa, last changes done:

- Reword commit history renaming the module
- Added the field  `warehouse_ids` to the method `_get_invalidation_fields` to invalidate the cache 
- Added the user warehouse filtered to avoid the error in products view

carlos-lopez-tecnativa reviewed (changes requested):
> I think it might be necessary to rewrite the entire Git history, or is it not needed, @pedrobaeza? Something like:
> 
> ```
> git filter-branch --tree-filter 'if [ -d old_module_name ]; then mv old_module_name new_module_name; fi' HEAD
> git rebase origin/branch
> ```

This is still pending. If you review the first commit, the module name is the old one. Did you apply the command I suggested

<img width="1039" height="400" alt="image" src="https://github.com/user-attachments/assets/5a0d6183-a66b-4e01-838e-0c315c9d9f20" />

christian-ramos-tecnativa commented:
Oh sorry @carlos-lopez-tecnativa, I thought the command was just for commit renaming, but now I see is way more complex than that
Thanks!

carlos-lopez-tecnativa reviewed (approved):
LGTM

pedrobaeza commented:
/ocabot merge nobump
